r2-core/src/main/java/com/linkedin/r2/transport/http/client/ratelimiter/RateLimiterExecutionTracker.java

```diff
@@ -53,4 +53,9 @@
      * @return maximum number of callbacks that can be stored in the RateLimiter
      */
     int getMaxBuffered();
+
+    /**
+     * @return amount of delay to be incurred before executing the next callback, based on provided rate
+     */
+    int getNextExecutionDelay(Rate rate);
```

Comment:
We can implement `return 0` as the default method here to make the interface backward compatible. And also, we don't need to override in the `SmoothRateLimiter` class

Reply:
Good idea